src/main/kotlin/com/sourcegraph/cody/chat/ChatSession.kt

```diff
@@ -236,7 +254,29 @@ private constructor(private val project: Project, newSessionId: CompletableFutur
     @RequiresEdt
     fun createNew(project: Project): AgentChatSession {
       val sessionId = createNewPanel(project) { it.server.chatNew() }
-      val chatSession = AgentChatSession(project, sessionId)
+      val chatSession = getSession(sessionId.get()) ?: AgentChatSession(project, sessionId)
```

Comment:
We are using this construct in 3 or 4 places already, across two classes.
Can we wrap it in a method?
Could be something like:
`getOrCreateSession(sessionId, () -> AgentChatSession): AgentChatSession`

Reply:
I agree. Fixed!